### PR TITLE
fix: noisy logs, no commit found for sha

### PR DESCRIPTION
**Pull request by yoshwata**
## Context

Such errors occurs in a situation, and it is noisy in any investigation or analysis.
```
Getting errors with [{"action":"getCommit","token":"248ec46cbf868bfe999bf6590c0ba42ee8cd90a2","params":{"owner":"yoshwata-test","repo":"branch-filter-test","ref":"0000000000000000000000000000000000000000"}}]: HttpError: No commit found for SHA: 0000000000000000000000000000000000000000
```

The situation is
- When a branch is deleted
- the branch is match with the branch filter regex.

`00000` means that deleted branch is not exist anymore.

~~Test will fail before  https://github.com/screwdriver-cd/data-schema/pull/410 is applied.~~

## Objective

API does not emit `No commit found for SHA: 0000~` error.

## References

https://github.com/screwdriver-cd/data-schema/pull/410
https://github.com/screwdriver-cd/scm-github/pull/179

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1343,6 +1344,7 @@ class GithubScm extends Scm {
                 lastCommitMessage: hoek.reach(webhookPayload, 'head_commit.message') || '',
                 hookId,
                 scmContext: scmContexts[0],
+                deleted,
```
Comment (kumada626) [inline index.js:1347]: Why parseHook returns this object when deleted is `true`? Isn't it enough to return `null` when deleted is true?

Comment (yoshwata) [reply index.js]: Thanks, I fixed

**Comment by yoshwata** [comment]
I need to fix the test

**Comment by yoshwata** [comment]
Hi, does anyone knows how to get x-github-sha value like [this](https://github.com/screwdriver-cd/scm-github/blob/83eed0230df8c6de489eec99c2edf3d5f3d12c95/test/index.test.js#L1333)?

**Comment by yoshwata** [comment]
I've fixed the test.

**Review by kumada626 — APPROVED** [review]
LGTM. Please change PR title to remove prefix number.

**Review by tkyi — APPROVED** [review]

**Pull request merged**